Gobabygo/Gobabygo.ino

```diff
@@ -36,65 +36,63 @@ float correctForDeadzone(float axis, float deadzone) {
   }
 }
 
-
 // This funtion gradually accelerates a value toward a target
 float accelerateToValue(float currentVal, float targetVal, int timeElapsed, float acceleration) {
   return currentVal + (targetVal - currentVal) * timeElapsed * acceleration;
 }
 
-
 void setup() {
   // put your setup code here, to run once:
   Serial.begin(9600);
   pinMode(rightMotorPin, OUTPUT);
   pinMode(leftMotorPin, OUTPUT);
 }
 
-
 void loop() {
   // put your main code here, to run repeatedly:
   currentTime = millis();
   timeSincePrint += currentTime - previousTime;
   timeSinceAcceleration = currentTime - previousTime;
   previousTime = currentTime;
   
+  float rawJoyXValue = analogRead(joyStickXPin);
   joyXVal = analogRead(joyStickXPin);
-  joyXVal = 2/1023 * joyXVal - 1;
+  joyXVal = 2.0/1023.0 * joyXVal - 1;
```

Comment:
To reduce number of calls to analogRead you could remove line 59 and modify like 60 to scale the raw x value. Same could be done for Y